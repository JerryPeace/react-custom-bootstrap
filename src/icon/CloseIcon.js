import PropTypes from 'prop-types';
import React from 'react';
import classNames from 'classnames';

class CloseIcon extends React.Component {

    static propTypes = {
        onClick: PropTypes.func
    };

    constructor (props) {
        super(props);
        this.onClick = ::this.onClick;
    }

    onClick(e) {
        e.preventDefault();
        !!this.props.onClick && this.props.onClick(e);
    }

    render() {
        return (
            <a
                href='#'
                className={classNames('close', this.props.className)}
                onClick={this.onClick}>
                <span>&times;</span>
            </a>
        );
    }
}

export default CloseIcon;
