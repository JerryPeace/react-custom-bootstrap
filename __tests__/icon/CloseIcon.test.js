import React from 'react';
import { mount } from 'enzyme';
import CloseIcon from 'icon/CloseIcon';

describe('CloseIcon', () => {
    let onClick;
    let target;

    beforeEach(() => {
        onClick = jest.fn();
        target = mount(<CloseIcon onClick={onClick}/>);

    });

    test('render CloseIcon correctly', () => {
        expect(target).toMatchSnapshot();
    });

    test('click on link', () => {
        target.find('a.close').simulate('click');
        expect(onClick.mock.calls.length).toBe(1);
    });
});
