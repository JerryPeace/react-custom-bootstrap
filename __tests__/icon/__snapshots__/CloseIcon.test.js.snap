// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseIcon render CloseIcon correctly 1`] = `
<CloseIcon
  onClick={[Function]}
>
  <a
    className="close"
    href="#"
    onClick={[Function]}
  >
    <span>
      Ã—
    </span>
  </a>
</CloseIcon>
`;
